api/app/Services/Availability/AvailabilityService.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services\Availability;
+
+use App\Contracts\AvailabilityServiceInterface;
+use App\Entity\Availability;
+use App\Exceptions\Availability\AvailabilityValidationException;
+use Carbon\Carbon;
+use Illuminate\Support\Collection;
+
+final class AvailabilityService implements AvailabilityServiceInterface
+{
+    private const MIDNIGHT_TIME = "00:00:00";
+    public function validateAvailabilities(Collection $availabilities, int $duration)
+    {
+        foreach ($availabilities as $availability) {
+            $this->validateAvailability($availability, $duration);
+        }
+        $this->checkAvailabilitiesOnOverlapping($availabilities);
+        return true;
+    }
+
+    private function validateAvailability(Availability $availability, int $duration)
+    {
+        $startDate = explode(" ", $availability->start_date)[0];
+        $endDate = explode(" ", $availability->end_date)[0];
+        $endTime = explode(" ", $availability->end_date)[1];
+
+        if ($availability->start_date > $availability->end_date) {
+            throw new AvailabilityValidationException(400, "Your end time cannot be before your start time!");
```

Comment:
The status code is not changing, you can remove it from the constructor, or use custom codes for being able to classify and use an appropriate translation for it on the client.